### DISCUSSION TITLE
Unable to render matplotlib widgets correctly with Jupyter Notebook Renderers

### DISCUSSION THREAD

**Discussion by SNOWBEAR-SCI** [microsoft/vscode-jupyter#16342] [Questions and Answers] [answered]
I don't know if this is the issue with the extension itself or if I am using it incorrectly.
The example code of slider from matplotlib can illustrate this issue:
https://matplotlib.org/stable/gallery/widgets/slider_demo.html#sphx-glr-gallery-widgets-slider-demo-py

If I run the ipynb with Jupyter notebook or the python file with vscode, the sliders and other matplotlib widgets are interactive. However, if I run the ipynb with vscode, I only get a non-interactive png output. In other words, the renderer extension does not output the interactive widgets as expected.

For kernel, I am using python 3.12 environment from anaconda in both Jupyter notebook and vscode. Most matplotlib and numpy functions except the widgets work, so I suppose the issue is with the renderer instead of kernel, function import, or other extensions.

**Comment by DonJayamanne** [comment, marked as answer]
To get interactive outputs in notebooks, you need to run the following magic `%matplotlib widget`
https://matplotlib.org/stable/users/explain/figure/interactive.html#jupyter-notebooks-jupyterlab

I'm surprised it works in Jupyter, I've tried Jupyter Lab and get the exact same output in VS Code as well (i.e. a non-interactive image).
Running the above magic before running the cell works as expected.
Not you need to install `ipympl`, see the link.

**Reply by SNOWBEAR-SCI** [reply]
Thanks! It works after installing ipympl for conda and adding `%matplotlib widget`. Maybe Jupyter has different preset?

**Reply by DonJayamanne** [reply]
Same for Jupyter too, the instructions are from the Matplotlib for Jupyter lab.
Possible you have some code (python package installed) thats configure some startup scripts and changed some of the configurations.
Either way, glad its resolved.

**Comment by Pierre-Bartet** [comment]
Hasen't been working for me for months despite multiple updates and reinstallations.
